src/C-interface/dense/bml_diagonalize_dense.c

```diff
@@ -333,9 +334,15 @@ bml_diagonalize_dense_single_complex(
 #ifdef NOBLAS
     LOG_ERROR("No BLAS library");
 #else
-    C_CHEEVR("V", "A", "U", &A->N, A_copy, &A->N, NULL, NULL, NULL, NULL,
-             &abstol, &M, evals, evecs, &A->N, isuppz, work, &lwork, rwork,
-             &lrwork, iwork, &liwork, &info);
+    {
+        float vl = -FLT_MAX;
+        float vu = FLT_MAX;
+        int il = 1;
+        int iu = A->N;
+        C_CHEEVR("V", "A", "U", &A->N, A_copy, &A->N, &vl, &vu, &il, &iu,
```

Comment:
Why do we need the 4 parameters `vi`, `vu`, `il`, `iu`? According to the documentation for `CHEEVR` [1] those are only used if the `RANGE` parameter (the second one) is `V` or `I`.

[1] http://www.netlib.org/lapack/explore-html/d9/de3/group__complex_h_eeigen_gac02f8a55f35ec7767a36b5a9f3ebb0d8.html

Reply:
Sorry for the delay, I didn't receive an email notification about the response.

This change is required for compatibility with Cray LibSci. Passing NULL throws a SIGSEGV when calling zheevr_  or cheevr_. We just need to define something sensible for compatibility, it should be harmless otherwise. Thanks.

Reply:
@nicolasbock Is this issue resolved? Any remaining issues to be addressed?

Reply:
Thanks for the information @mewall . Yes, that explains it.